registers.py

```diff
@@ -350,6 +357,22 @@ def write_cheader( fd, registers ):
                 for v in f.values:
                     definitions += v.to_c_definitions(prefix)
 
+                field_value_str = f"((value) & {sympy_to_c(mask.body())}) >> {sympy_to_c(offset.body())}"
+                if len(f.values):
+                    format_string += [f"\"{f.name}=%s\""]
+                    get_name_from_value = [
+                            f"({field_value_str} == {str_or_sympy_to_c(v.value)}) ? \"{v.name}\" :"
+                            for v in f.values if v.value
+                            ] + ["\"unknown\""]
```

Comment:
```suggestion
                            ] + ['"unknown"']
```

Reply:
Ideally we'd show the numeric value instead of "unknown". I don't know how to do that reasonably inside a macro.

Reply:
Addressed.
The issue with unknown values is addressed here: https://github.com/riscv/riscv-openocd/pull/892/files#diff-4187474b258a4ed647428725e3d087bc7f60b0e8a2648270beca34dcaa76498cR40